flask_caching/backends/base.py

```diff
@@ -28,19 +33,47 @@ def iteritems_wrapper(mappingorseq):
     return mappingorseq
 
 
+def extract_serializer_args(data):
+    result = dict()
+    serializer_prefix = "serializer_"
+    for key in tuple(data.keys()):
+        if key.startswith(serializer_prefix):
+            result[key] = data.pop(key)
+    return result
+
+
 class BaseCache(object):
     """Baseclass for the cache systems.  All the cache systems implement this
     API or a superset of it.
 
     :param default_timeout: The default timeout (in seconds) that is used if
                             no timeout is specified on :meth:`set`. A timeout
                             of 0 indicates that the cache never expires.
+    :param serializer_impl: Pickle-like serialization implementation. It should
```

Comment:
I'd much rather have a config variable like ``CACHE_SERIALIZER`` which lets me configure the serializer instead of setting it here

Reply:
For this variable `CACHE_SERIALIZER` we would have two values: `pickle` and `json` (and it's up to Flask-Caching internal implementation to reference the proper serializer object)?

Or do you want to have the serializer object directly set in the configuration? But I don't know if it's a good way to handle Flask configuration (we would not be able to load it from a json in this case).

Reply:
I will add `CACHE_SERIALIZER` and `CACHE_SERIALIZER_ERROR` config variables with dynamic (string-based) import support, but I think we should keep dependency injection in the base class, otherwise we will need to handle serialization in each implementation.
What about default value? Should we keep pickle as default serializer?

Reply:
I'd say we stay with 'pickle', issue a warning that this serializer is insecure and starting with 2.0 we switch to the JSON serializer.